subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -350,7 +350,9 @@ public void setExtensionDirs(@Nullable String extensionDirs) {
     @Internal
     public List<String> getAllCompilerArgs() {
         ImmutableList.Builder<String> builder = ImmutableList.builder();
-        builder.addAll(getCompilerArgs());
+        for (CharSequence arg : getCompilerArgs()) {
```

Comment:
I think we need some comments here, at least the Github issue url. In the subsequent maintainers' eyes, this is an unnecessary so they're likely to "optimize" this back to `builder.addAll(getCompilerArgs());`.

P.S. Why there's `CharSequence` instead of `Object`?

Reply:
That's what we have tests for. Someone else might change it, but the unit test will immediately tell them that this was wrong.

We could also have a helper method like `stringize`. Anything, but not a comment ;)

Reply:
Yeah. So why isn't it `for (Object arg : getCompilerArgs())`?

Reply:
Yeah it should be and I think we have a helper method for that already. I'll change it to that.